packages/expo/src/__tests__/ScreenOrientation-test.ts

```diff
@@ -0,0 +1,103 @@
+import { NativeModules } from 'react-native';
+import { mockPlatformAndroid, mockPlatformIOS, unmockAllProperties } from 'jest-expo';
+
+import * as ScreenOrientation from '../ScreenOrientation/ScreenOrientation';
+
+it (`throws when lockAsync is called with an unsupported type`, async () => {
+    await expect(ScreenOrientation.lockAsync(NaN as any)).rejects.toThrowError(TypeError);
+    await expect(ScreenOrientation.lockAsync('test' as any)).rejects.toThrowError(TypeError);
+});
+
+it (`NativeModules.lockAsync is not called with known unsupported locks`, async () => {
+    await ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.OTHER);
+    expect(NativeModules.ExpoScreenOrientation.lockAsync).not.toHaveBeenCalled(); 
+});
+
+it (`throws when supportsOrientationLockAsync is called with an unsupported type`, async () => {
+    await expect(ScreenOrientation.supportsOrientationLockAsync(NaN as any)).rejects.toThrowError(TypeError);
+    await expect(ScreenOrientation.supportsOrientationLockAsync('test' as any)).rejects.toThrowError(TypeError);
+});
+
+it (`throws when addOrientationChangeListener is called with an unsupported type`, async () => {
+    expect(()=>ScreenOrientation.addOrientationChangeListener(NaN as any)).toThrow(TypeError);
+    expect(()=>ScreenOrientation.addOrientationChangeListener('test' as any)).toThrow(TypeError);
+});
+
+it (`throws when removeOrientationChangeListener is called with an unsupported type`, async () => {
+    expect(()=>ScreenOrientation.removeOrientationChangeListener(NaN as any)).toThrow(TypeError);
+    expect(()=>ScreenOrientation.removeOrientationChangeListener('test' as any)).toThrow(TypeError);
+    expect(()=>ScreenOrientation.removeOrientationChangeListener({} as any)).toThrow(TypeError);
+});
+
+describe(`Android screen orientation`, () => {
+    beforeAll(() => {
```

Comment:
Use beforeEach / afterEach so that if a test modifies other properties, we get a clean slate in between tests